src/main/default/lwc/lookup/lookup.js

```diff
@@ -143,26 +166,57 @@ export default class Lookup extends LightningElement {
         this.updateSearchTerm(event.target.value);
     }
 
+    handleKeyDown(event) {
+        if (this.focusedResultIndex === null) {
+            this.focusedResultIndex = -1;
+        }
+        if (event.keyCode === ARROW_DOWN) {
+            // If we hit 'down', select the next item, or cycle over.
+            this.focusedResultIndex++;
+            if (this.focusedResultIndex >= this.searchResults.length) {
+                this.focusedResultIndex = 0;
+            }
+        } else if (event.keyCode === ARROW_UP) {
+            // If we hit 'up', select the previous item, or cycle over.
+            this.focusedResultIndex--;
+            if (this.focusedResultIndex < 0) {
+                this.focusedResultIndex = this.searchResults.length - 1;
+            }
+        } else if (event.keyCode === ENTER && this.hasFocus && this.focusedResultIndex >= 0) {
+            // If the user presses enter, and the box is open, and we have used arrows,
+            // treat this just like a click (add the item to selection and close the list)
+            this.addSelectedItem(this.searchResults[this.focusedResultIndex]);
+            this.hasFocus = false;
```

Comment:
I would also include one more line here:
event.currentTarget.blur();
Otherwise you will be able to reproduce a bug when multiselect is true.
When you hit enter on given record, suggestion box will disappear, but the caret will remain inside input.
You will try two options:
1. Start typing new search term nothing will happen.
2. I will click once again on input, but handle focus will not be executed.
Since I do not want to click outside of input and then click back again,  I need to make sure event.currentTarget.blur() is performed.

Reply:
ofc this will mean that user will have to click on input anyways to search for next value in multiselect, but I found this to be the easiest way to fix this bug. I do not have any other idea for transitioning from one selection to another.
Hope you will find above suggestions useful.
@pozil @rgalanakis